kratos/utilities/math_utils.h

```diff
@@ -464,243 +408,239 @@ class MathUtils
     }
 
     /**
-     * It inverts matrices of order 2, 3 and 4
-     * @param InputMatrix Is the input matrix (unchanged at output)
-     * @param InvertedMatrix Is the inverse of the input matrix
-     * @param InputMatrixDet Is the determinant of the input matrix
+     * @brief It inverts matrices of order 2, 3 and 4
+     * @param rInputMatrix Is the input matrix (unchanged at output)
+     * @param rInvertedMatrix Is the inverse of the input matrix
+     * @param rInputMatrixDet Is the determinant of the input matrix
      */
-
+    template<class TMatrix1, class TMatrix2>
```

Comment:
Why are the these templates here ? to allow both sparse and dense matrices ??

Reply:
To allow bounded and unbounded

Reply:
I needed this! Thanks.

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/35867376-e2292dfa-0b59-11e8-8ec9-db7ad94a9d5c.png)